go/vt/vtgate/scatter_conn.go

```diff
@@ -187,6 +188,14 @@ func (stc *ScatterConn) ExecuteMultiShard(
 				opts = session.Session.Options
 			}
 
+			if fetchLastInsertID {
+				if opts == nil {
+					opts = &querypb.ExecuteOptions{FetchLastInsertId: fetchLastInsertID}
+				} else {
+					opts.FetchLastInsertId = fetchLastInsertID
+				}
+			}
+
```

Comment:
This will not reset the `FetchLastInsertId` value.
```suggestion
if opts != nil {
    opts.FetchLastInsertId = fetchLastInsertID
} else if fetchLastInsertID {
    opts = &querypb.ExecuteOptions{FetchLastInsertId: fetchLastInsertID}
}
```

Reply:
We should add a test as well to validate if reset is happening correctly.